### PR TITLE
CPLP-1470 refactor owncompany search

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-1470

### CODE REVIEW

**Comment by jjeroch** [comment]
@qxz2mqe somehow in the pr's phil is not added as reviewer. Please always add Philip

File: src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs [hunk 1]
```diff
@@ -121,7 +121,7 @@ public UserRepository(PortalDbContext portalDbContext)
                 && companyUserId.HasValue ? companyUser.Id == companyUserId!.Value : true
                 && firstName != null ? companyUser.Firstname == firstName : true
                 && lastName != null ? companyUser.Lastname == lastName : true
-                && email != null ? companyUser.Email == email : true);
+                && email != null ? EF.Functions.ILike(companyUser.Email!, $"%{email}%") : true);
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs:124]: We should think about decoding the value in email for safety reasons to make sure one can't query more than he should and prevent sql injection.

@jjeroch is that required? if no i would approve the pr

Comment (Phil91) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs:124]: @qxz2mqe an example for escaping would be:

`
    public IQueryable<CompanyUser> GetOwnCompanyAppUsersUntrackedAsync(
        Guid appId,
        string iamUserId,
        string? firstName = null,
        string? lastName = null,
        string? email = null,
        string? roleName = null) {
        char[] escapeChar = { '%', '_', '[', ']', '^' };
        return _context.CompanyUsers.AsNoTracking()
            .Where(companyUser => companyUser.UserRoles.Any(userRole => userRole.Offer!.Id == appId) && companyUser.IamUser!.UserEntityId == iamUserId)
            .SelectMany(companyUser => companyUser.Company!.CompanyUsers)
            .Where(companyUser => firstName == null || EF.Functions.ILike(companyUser.Firstname!, $"%{firstName.Trim(escapeChar)}%")
                && lastName == null || EF.Functions.ILike(companyUser.Lastname!, $"%{lastName!.Trim(escapeChar)}%")
                && email == null || EF.Functions.ILike(companyUser.Email!, $"%{email!.Trim(escapeChar)}%")
                && roleName == null || companyUser.UserRoles.Any(userRole => EF.Functions.ILike(userRole.UserRoleText, $"{roleName!.Trim(escapeChar)}%")));
        }
`

Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/UserRepository.cs]: done

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**